lib/sessions.js

```diff
@@ -138,16 +138,42 @@ class ServerSessionPool {
     }
 
     this.topology = topology;
-    this.sessions = [];
+    this.pool = [];
```

Comment:
I'm not sure `pool` is a more descriptive name, it certainly isn't in the unit tests lower in the file where we are checking `pool.pool` (also we could avoid about half the changes in this PR if we just leave this as `sessions` 😄)